src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -203,6 +221,90 @@ public override void Return(T[] array, bool clearArray = false)
             }
         }
 
+        public bool Trim()
+        {
+            uint milliseconds = (uint)Environment.TickCount;
+
+            MemoryPressure pressure = GetMemoryPressure();
+            foreach (PerCoreLockedStacks bucket in _buckets)
```

Comment:
I would put another logging message called TrimPoll(int milleseconds, MemoryPressure pressure).  

This lets you diagnose the case where things are NOT being trimmed and you don't know why.   Thus you want to know all the information used to decide to trim or not (and the fact that the callback happened).

Reply:
Ok